### PR TITLE
Fix/file io wrapper

### PR DESCRIPTION
**Pull request by TroubleMakerBen**
Main objective: fix issue where non-sandboxed apps on mac might see other apps telemetry data. FIXED

What also happened: refactored file i/o into category. Could need more work, as some of the crash functionality could move from full NSURL to path components but that seemed a bit much for a bugfix.

what's missing:
More unit tests. MSCrashesTests and MSCrashesUtilTests already test the refactored logic indirectly but probably gonna do more testes.

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashesTests/MSCrashesTests.mm [hunk 1]
```diff
@@ -243,7 +238,9 @@ - (void)testProcessCrashes {
 
   // Then
   assertThat(self.sut.crashFiles, hasCountOf(1));
-  assertThatLong([self.sut.fileManager contentsOfDirectoryAtPath:crashesPath error:nil].count, equalToLong(1));
+
```
Comment (clpolet) [inline AppCenterCrashes/AppCenterCrashesTests/MSCrashesTests.mm]: Can probably remove this space.

File: AppCenterCrashes/AppCenterCrashesTests/MSCrashesTests.mm [hunk 2]
```diff
@@ -405,64 +405,54 @@ - (void)testSetupLogBufferWorks {
...
   [self.sut setupLogBuffer];
 
   // Then
-  NSArray *second = [[NSFileManager defaultManager]
-      contentsOfDirectoryAtPath:reinterpret_cast<NSString *_Nonnull>([self.sut.logBufferDir path])
-                          error:&error];
+  NSArray *second = [MSUtility contentsOfDirectory:self.sut.logBufferPathComponent propertiesForKeys:nil];
   for (int i = 0; i < ms_crashes_log_buffer_size; i++) {
-    XCTAssertTrue([first[i] isEqualToString:second[i]]);
+    XCTAssertTrue([([first[i] absoluteString] ?: @"") isEqualToString:([second[i] absoluteString] ?: @"")]);
```
Comment (clpolet) [inline AppCenterCrashes/AppCenterCrashesTests/MSCrashesTests.mm]: What's the reason behind this change?

File: AppCenterCrashes/AppCenterCrashesTests/MSCrashesUtilTests.m [hunk 3]
```diff
@@ -29,32 +30,78 @@ - (void)tearDown {
...
+
+  // If
   NSString *expectedDir;
-  XCTAssertNotNil(crashesDir);
+#if TARGET_OS_TV
+  expectedDir = @"/Library/Caches/com.microsoft.appcenter/crashes";
+#else
 #if TARGET_OS_OSX
-  expectedDir = @"/Library/Caches/com.test.app/com.microsoft.appcenter/crashes";
+  expectedDir = @"/Library/Application%20Support/(null)/com.microsoft.appcenter/crashes";
```
Comment (clpolet) [inline AppCenterCrashes/AppCenterCrashesTests/MSCrashesUtilTests.m]: I think it takes the info.plist from the test target, have you double checked if the app bundle ID was there? It can be missing.

Comment (TroubleMakerBen) [reply AppCenterCrashes/AppCenterCrashesTests/MSCrashesUtilTests.m]: the app bundle is `nil` for test targets. It took me a while to figure that one out last night.

**Comment by TroubleMakerBen** [comment]
Merging this. 👍
